### PR TITLE
Update deps and no error for failed deploy

### PR DESCRIPTION
**Pull request by jgodson**
The primary goal of this PR is to avoid the need for actions read permissions that by default we haven't requested in the oxygen deployment workflow files. This made failed Oxygen deployments not great, like so:

image

Now they will be something more obvious, like this:
image

This also avoids deployments on a PR staying in a `Pending` status even though the workflow failed.
image

They now show up as `Failure`
image

The one non-optimal thing about this PR is that we don't provide the perfect link to the job as that's not available, we need to be able to read actions to grab the jobs and compare the job name to do that unfortunately. The best we can do is link to the workflow run (screenshot above). I think it's a much better experience by default though than it was.

This is a link with the workflow previously with `actions: read` permissions
image

This is after the PR (no need for extra permissions because there's no API request for it)
image

### What else does this do?

I updated the dependencies as well (which is why there are so many files changes). I also removed the deprecated way of setting output in the readme and CI files. The CI needed some tweaks to support the new way this works (ie: we wanted the action to always do `core.setFailed` to fail the workflow)

### Testing

You can test by using this action in an existing oxygen workflow file by changing it from `v1` to `v1.0.3`.

### CODE REVIEW

**Review by filipedeo — APPROVED** [review]
Tested with a test shop and the `v1.0.3` flow is looking good 👌

**Review by jsaubry — APPROVED** [review]
Tried with `v1.0.3` error and success and it seems good.

However, to be honest, even by setting `v1.0.3` I'm not 100% confident that it used that version. Is there a way to me to tell that my test didn't test v1?

**Comment by jgodson** [comment]
> However, to be honest, even by setting v1.0.3 I'm not 100% confident that it used that version. Is there a way to me to tell that my test didn't test v1?

It shows which version it uses in the logs

image

File: index.js [hunk 1]
```diff
@@ -94,12 +57,24 @@ async function main() {
...
 }
 
-main().catch(function (error) {
+main().then((result) => {
+  if (process.env.WORKFLOW_CI) {
+    const expected = process.env.EXPECTED_RESULT
+    if (result != expected) {
+      core.setFailed(`Expected ${expected} but got ${result}`);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (jgodson) [reply index.js]: Going to merge as is to fix the unwanted behaviour , but happy to change in a follow up if you feel it can be improved further 👍🏻

**Pull request merged**